prompt2model/utils/parse_json_responses.py

```diff
@@ -31,33 +31,30 @@ def extract_response(
     try:
         response_json = json.loads(response_text, strict=False)
     except json.decoder.JSONDecodeError:
-        # TODO: Should error be thrown here?
         logger.warning(f"API response was not a valid JSON: {response_text}")
         return None
 
     missing_keys = [key for key in required_keys if key not in response_json]
     if len(missing_keys) != 0:
-        # TODO: Should error be thrown here?
         logger.warning(f'API response must contain {", ".join(required_keys)} keys')
         return None
 
     final_response = {}
     for key in required_keys + optional_keys:
         if key not in response_json:
-            continue  # optional key, we checked for required keys before
+            continue  # optional key, we checked for required keys before.
```

Comment:
1) I think this comment is a bit unclear
2) I feel this is not a great situation for an inline comment - it looks a little off.

```suggestion
            # This is an optional key, so exclude it from the final response.
            continue
```